block-streamer/src/bitmap.rs

```diff
@@ -0,0 +1,367 @@
+            last_bit_index: first_bit_index + zero_count,
+        }
+    }
+
+    fn decompress_bitmap(&self, compressed_bitmap: &[u8]) -> Vec<u8> {
+        let compressed_bit_length: usize = compressed_bitmap.len() * 8;
+        let mut current_bit_value: bool = (compressed_bitmap[0] & 0b10000000) > 0;
+        let mut decompressed_byte_array: Vec<u8> = Vec::new();
+
+        let mut compressed_bit_index = 1;
+        let mut decompressed_bit_index = 0;
+
+        while compressed_bit_index < compressed_bit_length {
+            let decoded_elias_gamma =
+                self.decode_elias_gamma_entry(compressed_bitmap, compressed_bit_index);
+            if decoded_elias_gamma.value == 0 {
+                break;
+            }
+
+            compressed_bit_index = decoded_elias_gamma.last_bit_index + 1;
+            let mut bit_index_offset: usize = 0;
+            while current_bit_value && (bit_index_offset < decoded_elias_gamma.value) {
+                while decompressed_bit_index + bit_index_offset
+                    >= (decompressed_byte_array.len() * 8)
+                {
+                    decompressed_byte_array.push(0b00000000);
+                }
+                self.set_bit(
+                    &mut decompressed_byte_array,
+                    decompressed_bit_index + bit_index_offset,
+                    true,
+                    true,
+                );
+                bit_index_offset = bit_index_offset + 1;
+            }
+
+            decompressed_bit_index += decoded_elias_gamma.value;
+            current_bit_value = !current_bit_value;
+        }
+
+        decompressed_byte_array
+    }
+
+    fn merge_compressed_bitmap_into_base_bitmap(
+        &self,
+        base_bitmap: &mut Bitmap,
+        compressed_bitmap: &Bitmap,
+    ) -> anyhow::Result<()> {
+        let decompressed_add_bitmap: Vec<u8> = self.decompress_bitmap(&compressed_bitmap.bitmap);
+        let start_bit_index: usize = match compressed_bitmap
+            .start_block_height
+            .checked_sub(base_bitmap.start_block_height)
+        {
+            Some(result) => result,
+            None => {
+                return Err(anyhow!(
+                    "Start block height in bitmap was lower than provided lowest block height",
+                ))
+            }
+        };
+
+        for bit_index_offset in 0..(decompressed_add_bitmap.len() * 8) {
+            let decompressed_bit_value = self.get_bit(&decompressed_add_bitmap, bit_index_offset);
+            while start_bit_index + bit_index_offset >= base_bitmap.bitmap.len() * 8 {
+                base_bitmap.bitmap.push(0b00000000);
+            }
+
+            self.set_bit(
+                &mut base_bitmap.bitmap,
+                start_bit_index + bit_index_offset,
+                decompressed_bit_value,
+                false,
+            );
+        }
+
+        Ok(())
+    }
+
+    pub fn get_merged_bitmap(
```

Comment:
```suggestion
    pub fn merge_bitmaps(
```
Nit: this seems more clear?

Reply:
I'll go with `merge_compressed_bitmaps`. This way, it's clear this is taking in compressed bitmaps, and the argument type Base64Bitmap will also indicate this requires a undecoded compressed bitmap too.